### PR TITLE
Task1

### PR DESCRIPTION
**Pull request by Paul044**
Task https://epa.ms/nodejs19-hw1

### CODE REVIEW

File: task1/task2.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        return lineContent;
+      })
+      .subscribe((jsonObj) => {
+        return new Promise((resolve, reject) => {
+          delete jsonObj.amount;
+          resolve();
+        });
+      })
+  )
+  .on("error", onError)
```
Comment (meteorra) [inline task1/task2.js]: It's enough to use error callback only once

File: task1/task2.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  .pipe(
+    csv()
+      .preFileLine((line, lineNumber) => {
+        let lineContent = lineNumber ? line : line.toLowerCase();
+        return lineContent;
+      })
+      .subscribe((jsonObj) => {
+        return new Promise((resolve, reject) => {
+          delete jsonObj.amount;
```
Comment (meteorra) [inline task1/task2.js]: Modifying incoming object is error prone. It's usually better to use new object with copied fields

File: task1/task2.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+const fs = require("fs");
+const csv = require("csvtojson");
+
+const csvFilePath = "./csv/nodejs-hw1-ex1.csv";
+const outputFilePath = "./output.txt";
+const readable = fs.createReadStream(csvFilePath);
+const writable = fs.createWriteStream(outputFilePath);
+const onError = (error) => console.log(error);
+
+readable
```
Comment (meteorra) [inline task1/task2.js]: You can use stream.pipeline for simplicity: https://nodejs.org/api/stream.html
E.g.:
```
pipeline(
  fs.createReadStream('archive.tar'),
  zlib.createGzip(),
  fs.createWriteStream('archive.tar.gz'),
  (err) => {
    if (err) {
      console.error('Pipeline failed.', err);
    } else {
      console.log('Pipeline succeeded.');
    }
  }
);

File: task1/task2.js (hunk 1 above)
Comment (Paul044) [reply task1/task2.js]: done

File: task1/task2.js (hunk 2 above)
Comment (Paul044) [reply task1/task2.js]: done

File: task1/task2.js (hunk 3 above)
Comment (Paul044) [reply task1/task2.js]: done

**Pull request merged**